server/graphql/v1/mutations/expenses.js

```diff
@@ -95,6 +95,26 @@ export async function updateExpenseStatus(remoteUser, expenseId, status) {
   return res;
 }
 
+/** Check expense's attachments values, throw if something's wrong */
+const checkExpenseAttachments = (expenseData, attachments) => {
+  // Check the number of attachments
+  if (!attachments || attachments.length === 0) {
+    throw new ValidationFailed({ message: 'Your expense needs to have at least one attachment' });
+  } else if (attachments.length > 100) {
+    throw new ValidationFailed({ message: 'Expenses cannot have more than 100 attachments' });
+  }
+
+  // Check amounts
+  const sumAttachments = attachments.reduce((total, attachment) => total + attachment.amount, 0);
+  if (sumAttachments !== expenseData.amount) {
+    throw new ValidationFailed({
+      message: `The sum of all attachments must be equal to the total expense's amount. Expense's total is ${expenseData.amount}, but the total of attachments was ${sumAttachments}.`,
+    });
+  }
+};
```

Comment:
Interesting.
This is basically a runtime integration test.
In my head, having the sum being calculated in the backend seemed simpler but if we take into consideration the frontend and third-party use of the API, I think this saves us some code/complexity.

Reply:
I would actually prefer to compute the amount from attachments dynamically as you suggested  and not store any amount in `Expense` but we have too much code relying on this column (queries for statistics for example) in too many places, the conversion seemed tedious and risky.

Reply:
Perfect.